src/controller/python/chip/ChipExceptions.py

```diff
@@ -0,0 +1,103 @@
+#
+#    Copyright (c) 2021 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+__all__ = [
+    "ChipStackException",
+    "ChipStackError",
+    "DeviceError",
+    "InvalidArgumentCount",
+    "InvalidArgumentType",
+    "UnknownCluster",
+    "UnknownCommand",
+]
+
+
+class ChipStackException(Exception):
+    pass
+
+
+class ChipStackError(ChipStackException):
+    def __init__(self, err, msg=None):
+        self.err = err
+        if msg != None:
```

Comment:
`if msg:` seems more pythonic. or maybe even:

```py
self.msg = msg if msg else "Chip Stack Error %d" % err
```

is "ld" required in python or can we just use %d? Would 0x%08X be easier to follow up in the code as our codes have hex encodings?